app/js/lib/foxbox/settings.js

```diff
@@ -22,178 +21,203 @@ const API_VERSION = 1;
+    key: 'watchInterval',
+    type: 'number',
+    defaultValue: 3000,
+  }),
+  ONLINE_CHECKING_INTERVAL: Object.freeze({
+    key: 'onlineCheckingInterval',
+    type: 'number',
+    defaultValue: 5000,
+  }),
+  ONLINE_CHECKING_LONG_INTERVAL: Object.freeze({
+    key: 'onlineCheckingLongInterval',
+    type: 'number',
+    defaultValue: 1000 * 60 * 5,
+  }),
+
+  // String settings.
+  LOCAL_ORIGIN: Object.freeze({ key: 'localOrigin' }),
+  TUNNEL_ORIGIN: Object.freeze({ key: 'tunnelOrigin' }),
+  CLIENT: Object.freeze({ key: 'client' }),
+  SESSION: Object.freeze({ key: 'session' }),
+  PUSH_ENDPOINT: Object.freeze({ key: 'pushEndpoint' }),
+  PUSH_PUB_KEY: Object.freeze({ key: 'pushPubKey' }),
+  REGISTRATION_SERVICE: Object.freeze({
+    key: 'registrationService',
+    defaultValue: 'https://knilxof.org:4443/ping',
+  }),
+});
 
 export default class Settings extends EventDispatcher {
-  constructor() {
-    super(['session', 'polling-setting']);
-
-    const localOrigin = storage.getItem(`${PREFIX}localOrigin`);
-
-    const pollingEnabled = storage.getItem(`${PREFIX}pollingEnabled`) !== null ?
-      storage.getItem(`${PREFIX}pollingEnabled`) === 'true' :
-      DEFAULT_POLLING_ENABLED;
-
-    this._configured = storage.getItem(`${PREFIX}configured`) !== null ?
-      storage.getItem(`${PREFIX}configured`) === 'true' : false;
-    this._localOrigin = localOrigin;
-    this._tunnelOrigin = storage.getItem(`${PREFIX}tunnelOrigin`) || '';
-    this._client = storage.getItem(`${PREFIX}client`) || '';
-    this._session = storage.getItem(`${PREFIX}session`);
-    this._skipDiscovery = storage.getItem(`${PREFIX}skipDiscovery`) === 'true';
-    this._pollingEnabled = pollingEnabled;
-    this._pushEndpoint = storage.getItem(`${PREFIX}push_endpoint`) || null;
-    this._pushPubKey = storage.getItem(`${PREFIX}push_pubKey`) || null;
-  }
+  constructor(storage = localStorage) {
+    super();
+
+    // Not all browsers have localStorage supported or activated.
+    this[p.storage] = storage || {
+      getItem: () => null,
+      setItem: () => {},
+      removeItem: () => {},
+      clear: () => {},
+    };
+
+    this[p.values] = new Map();
+
+    Object.keys(settings).forEach((settingName) => {
+      const setting = settings[settingName];
+      const settingStringValue = this[p.storage].getItem(
+        `${PREFIX}${setting.key}`
+      );
+
+      // Setting values directly to avoid firing events on startup.
+      this[p.values].set(
+        setting,
+        this[p.stringToSettingTypedValue](setting, settingStringValue)
+      );
+    });
 
-  clear() {
-    return new Promise((resolve) => {
-      // @todo Remove only the items set here.
-      storage.clear();
-      this._session = null;
+    window.addEventListener('storage', this[p.onStorage].bind(this));
 
-      resolve();
-    });
+    Object.seal(this);
   }
 
-  get configured() {
-    return this._configured;
+  get [settings.CONFIGURED.key]() {
```

Comment:
Yep, true, here is what I see:
![screenshot from 2016-05-11 14-14-13](https://cloud.githubusercontent.com/assets/1713708/15180609/ebc48a7a-1782-11e6-9f46-438181450be1.png)

So yeah, maybe let's just not use computed property names and use plain old names for getters and setters. It's small duplication, but benefits outweigh (docs and just much easier to read).

What do you think?